src/main/java/carpet/script/exception/ExpressionException.java

```diff
@@ -8,28 +8,32 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Supplier;
 
 /* The expression evaluators exception class. */
 public class ExpressionException extends RuntimeException implements ResolvedException
 {
     public final Context context;
     public final List<FunctionValue> stack = new ArrayList<>();
+    private final Supplier<String> lazyStacktrace;
+    private String cachedMessage = null;
 
     public ExpressionException(Context c, Expression e, String message)
     {
-        super(makeMessage(c, e, null, message));
-        context = c;
+        this(c, e, null, message);
     }
 
     public ExpressionException(Context c, Expression e, Tokenizer.Token t, String message)
     {
-        super(makeMessage(c, e, t, message));
+        super("Error");
+        lazyStacktrace = () -> makeMessage(c, e, t, message);
```

Comment:
And specific reason for deferring it?

Reply:
Well, the `makeMessage()` function calls `errorSnooper.apply()`, which directly notifies the source. I had trouble finding a way to change it not to throw in potentially handled ones, and this worked. There are probably many other ways to do it.

Reply:
ok, makes sense